var/spack/repos/builtin/packages/py-dictdiffer/package.py

```diff
@@ -12,9 +12,10 @@ class PyDictdiffer(PythonPackage):
     homepage = "https://github.com/inveniosoftware/dictdiffer"
     pypi = "dictdiffer/dictdiffer-0.8.1.tar.gz"
 
+    version("0.9.0", sha256="17bacf5fbfe613ccf1b6d512bd766e6b21fb798822a133aa86098b8ac9997578")
     version("0.8.1", sha256="1adec0d67cdf6166bda96ae2934ddb5e54433998ceab63c984574d187cc563d2")
 
     depends_on("python@2.7:2.8,3.4:", type=("build", "run"))
-    depends_on("py-setuptools", type=("build", "run"))
+    depends_on("py-setuptools", type=("build", "run"), when="@:0.8")
```

Comment:
```suggestion
    depends_on("py-setuptools", type="build", when="@0.9:")
    depends_on("py-setuptools", type=("build", "run"), when="@:0.8")
```
You still need it as build dependency though because, as far as I know, we don't rely on implicit dependencies.